docs/architecture/grpc-for-wcf-developers/authentication.md

```diff
@@ -0,0 +1,18 @@
+---
+title: Authentication and Authorization
+description: gRPC for WCF Developers | Authentication and Authorization
+author: markrendle
+ms.date: 09/02/2019
+---
+
+# Authentication and authorization
+
+gRPC authentication works on two levels. Channel-level authentication is based on SSL/TLS certificates and is applied when the channel is opened. Call-level authentication is handled using tokens which are applied in metadata when the call is made. You can use either or both of these mechanisms to secure your service.
```

Comment:
Hmm, I don't think I'd phrase it like that. `ChannelCredentials` is a thing in gRPC that can be used to apply per-call authentication.

There are many options for authn over gRPC: https://docs.microsoft.com/en-us/aspnet/core/grpc/authn-and-authz?view=aspnetcore-3.0#other-authentication-mechanisms. Client certificates is just one of those many options.

Reply:
Yeah, I think it might make more sense to put CallCredentials first and then make the point that ChannelCredentials can also include CallCredentials to be applied to every request.

@nishanil Would it be OK to link to the (excellent) docs.microsoft.com content on this subject from the book, and just include a precis and some sample code?

Reply:
Yes, please link to docs wherever appropriate. You can also have a "References" section where you can guide them to other materials that you think are useful to the reader in the context.